src/symbolic/symbolic_choice.ml

```diff
@@ -390,41 +401,41 @@ module Make (Thread : Thread_intf.S) = struct
     let* () = yield prio in
     let* thread in
     let* solver in
-    let pc = Thread.pc thread in
-    let sliced_pc = Symbolic_path_condition.slice pc v in
+    let pc = Thread.pc thread |> Symbolic_path_condition.slice_on_condition v in
     let stats = Thread.bench_stats thread in
     let reachability =
-      Benchmark.handle_time_span stats.solver_time @@ fun () ->
-      Solver.check solver sliced_pc
+      Benchmark.handle_time_span stats.solver_sat_time @@ fun () ->
+      Solver.check solver pc
     in
     return reachability
 
   let get_model_or_stop symbol =
     let* () = yield Prio.default in
     let* solver in
-    let+ thread in
-    let pc = Thread.pc thread |> Symbolic_path_condition.to_set in
+    let* thread in
```

Comment:
@krtab, I'm trying to convince myself that there was no particular reason why this was a `let+` before, could you convince me? (I could be convinced by a "I have no idea".)